lib/reply.js

```diff
@@ -192,9 +192,9 @@ function handleError (reply, error, cb) {
   reply.res.statusCode = statusCode
 
   if (statusCode >= 500) {
-    reply.res.log.error({ res: reply.res, err: error }, error && error.message)
+    reply.res.log.error({ req: reply.request.raw, res: reply.res, err: error }, error && error.message)
   } else if (statusCode >= 400) {
-    reply.res.log.info({ res: reply.res, err: error }, error && error.message)
+    reply.res.log.info({ req: reply.request.raw, res: reply.res, err: error }, error && error.message)
```

Comment:
I don't know if here is ok.

For 5xx errors is oki because they happen not so frequently.
For 4xx the errors are handled correctly by the framework or by the user logic.

Thoughts!

Reply:
If you use `setErrorHandler` you will handle also `5xx` errors.

Reply:
I don't understand what you mean.

5xx are not common as the 4xx errors are. 4xx means the server is able to handle it correctly. 5xx means the server aren't. For me only for the 5xx errors the request should be logged completely.

NB: this PR degrees the performance on 4xx errors.

Reply:
@fastify/fastify thoughts?

Reply:
Do we really need this feature at all? We provide a `setErrorHandler` hook.

```js
fastify.setErrorHandler(function (err, reply) {})
```

Reply:
@StarpTech I think we need this feature (maybe only on 5xx error), because if you do that in the interceptor you will:
- force user to add code on every project (debug a 5xx without request information can quickly be a nightmare)
- send 2 logs instead of a concise one (even if we have `reqId` to group those logs, it can be less easier to debug)

@allevo Debug a 4xx with request information can be very useful, but I don't know how It can negatively impact performance, not an easy choice!

Reply:
People can log errors however they want using `setErrorHandler`. If they don't want to send a custom response, they can forward the error back to Fastify:

```js
fastify.setErrorHandler((err, reply) => {
  // ... custom logging
  reply.send(err)
})
```

Although if they did that, all of [this code](https://github.com/fastify/fastify/blob/9234f865c432c815b7e69eaa1d356a47930ceeb7/lib/reply.js#L181-L202) would get run again (which would be unnecessary since it's the same `error`).